contracts/distribution.js

```diff
@@ -0,0 +1,301 @@
+/* eslint-disable no-await-in-loop */
+/* eslint no-underscore-dangle: ["error", { "allow": ["_id"] }] */
+/* global actions, api */
+
+// limit recipients per distribution batch
+const MAX_RECIPIENTS = 40;
+
+actions.createSSC = async () => {
+  const tableExists = await api.db.tableExists('batches');
+  if (tableExists === false) {
+    await api.db.createTable('batches');
+    await api.db.createTable('params');
+
+    const params = {};
+    params.distCreationFee = '500';
+    params.distUpdateFee = '250';
+    await api.db.insert('params', params);
+  }
+};
+
+actions.updateParams = async (payload) => {
+  if (api.sender !== api.owner) return;
+
+  const {
+    distCreationFee,
+    distUpdateFee,
+  } = payload;
+
+  const params = await api.db.findOne('params', {});
+
+  if (distCreationFee) {
+    if (!api.assert(typeof distCreationFee === 'string' && !api.BigNumber(distCreationFee).isNaN() && api.BigNumber(distCreationFee).gte(0), 'invalid distCreationFee')) return;
+    params.distCreationFee = distCreationFee;
+  }
+  if (distUpdateFee) {
+    if (!api.assert(typeof distUpdateFee === 'string' && !api.BigNumber(distUpdateFee).isNaN() && api.BigNumber(distUpdateFee).gte(0), 'invalid distUpdateFee')) return;
+    params.distUpdateFee = distUpdateFee;
+  }
+
+  await api.db.update('params', params);
+};
+
+async function processBatch(batch, symbol, isFlush = false) {
+  let balance;
+  let balanceInd;
+  let payout;
+
+  for (let i = 0; i < batch.tokenBalances.length; i += 1) {
```

Comment:
Suggestion, have balance store the actual object instead of just the quantity, and you can use it below without storing balanceInd also.

This can be shortened to

const balance = batch.tokenBalances.find(b => b.symbol === symbol);

Reply:
Thanks, much more elegant.